File: backend/src/main/java/com/zzang/chongdae/offering/service/dto/OfferingDetailResponse.java
package com.zzang.chongdae.offering.service.dto;

import com.zzang.chongdae.offering.domain.OfferingJoinedCount;
import com.zzang.chongdae.offering.domain.OfferingPrice;
import com.zzang.chongdae.offering.domain.OfferingStatus;
import com.zzang.chongdae.offering.repository.entity.OfferingEntity;
import java.time.LocalDateTime;

public record OfferingDetailResponse(Long id,
                                     String title,
                                     String productUrl,
                                     String meetingAddress,
                                     String meetingAddressDetail,
                                     String description,
                                     LocalDateTime meetingDate,
                                     Integer currentCount,
                                     Integer totalCount,
                                     String thumbnailUrl,
                                     Integer dividedPrice,
                                     Integer totalPrice,
                                     Integer originPrice,
                                     OfferingStatus status,
                                     Long memberId,
                                     String nickname,
                                     Boolean isProposer,
                                     Boolean isParticipated) {

    public OfferingDetailResponse(OfferingEntity offering,
                                  OfferingPrice offeringPrice,
                                  OfferingJoinedCount offeringJoinedCount,
                                  Boolean isProposer,
                                  Boolean isParticipated) {
        this(offering.getId(),
                offering.getTitle(),
                offering.getProductUrl(),
                offering.getMeetingAddress(),
                offering.getMeetingAddressDetail(),
                offering.getDescription(),
                offering.getMeetingDate(),
                offeringJoinedCount.getCurrentCount(),
                offering.getTotalCount(),
                offering.getThumbnailUrl(),
                offeringPrice.calculateDividedPrice(),
                offering.getTotalPrice(),
                offering.getOriginPrice(),
                offering.getOfferingStatus(),
                offering.getMember().getId(),
                offering.getMember().getNickname(),
                isProposer,
                isParticipated
        );
    }
}

File: android/app/src/main/java/com/zzang/chongdae/data/remote/dto/response/offering/OfferingDetailResponse.kt
package com.zzang.chongdae.data.remote.dto.response.offering

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
data class OfferingDetailResponse(
    @SerialName("id") val id: Long,
    @SerialName("title") val title: String,
    @SerialName("productUrl") val productUrl: String?,
    @SerialName("meetingAddress") val meetingAddress: String,
    @SerialName("meetingAddressDetail") val meetingAddressDetail: String,
    @SerialName("description") val description: String,
    @SerialName("meetingDate") val meetingDate: String,
    @SerialName("currentCount") val currentCount: Int,
    @SerialName("totalCount") val totalCount: Int,
    @SerialName("thumbnailUrl") val thumbnailUrl: String?,
    @SerialName("dividedPrice") val dividedPrice: Int,
    @SerialName("totalPrice") val totalPrice: Int,
    @SerialName("originPrice") val originPrice: Int?,
    @SerialName("status") val condition: RemoteOfferingStatus,
    @SerialName("isProposer") val isProposer: Boolean,
    @SerialName("nickname") val nickname: String,
    @SerialName("isParticipated") val isParticipated: Boolean,
)
